docs/01-app/03-api-reference/04-functions/cacheLife.mdx

```diff
@@ -35,79 +37,106 @@ const nextConfig = {
+- During this time, the client serves cached content immediately without a network request
+- After this period expires, the client checks with the server for updates
 
-### Custom cache profiles
+```tsx
+cacheLife({ stale: 300 }) // 5 minutes
+```
 
-You can configure custom cache profiles by adding them to the [`cacheLife`](/docs/app/api-reference/config/next-config-js/cacheLife) option in your `next.config.ts` file.
+#### `revalidate`
 
-Cache profiles are objects that contain the following properties:
+How often the server regenerates cached content in the background.
 
-| **Property** | **Value** | **Description**                                                                                                             | **Requirement**                             |
-| ------------ | --------- | --------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------- |
-| `stale`      | `number`  | Duration the client should cache a value without checking the server.                                                       | Optional                                    |
-| `revalidate` | `number`  | Frequency at which the cache should refresh on the server; stale values may be served while revalidating.                   | Optional                                    |
-| `expire`     | `number`  | Maximum duration for which a value can remain stale before switching to dynamic fetching; must be longer than `revalidate`. | Optional - Must be longer than `revalidate` |
+- When a request arrives after this period, the server:
+  1. Serves the cached version immediately (if available)
+  2. Regenerates content in the background
+  3. Updates the cache with fresh content
+- Similar to [Incremental Static Regeneration (ISR)](/docs/app/guides/incremental-static-regeneration)
 
-The "stale" property differs from the [`staleTimes`](/docs/app/api-reference/config/next-config-js/staleTimes) setting in that it specifically controls client-side router caching. While `staleTimes` is a global setting that affects all instances of both dynamic and static data, the `cacheLife` configuration allows you to define "stale" times on a per-function or per-route basis.
+```tsx
+cacheLife({ revalidate: 900 }) // 15 minutes
+```
 
-### `stale` time in the client router cache
+#### `expire`
 
-The "stale" property does not set the `Cache-control: max-age` header. Instead, it controls the client-side router cache. The server sends this value to the client via the `x-nextjs-stale-time` response header (in seconds), which the client router uses to determine how long to cache the route before needing to revalidate.
+Maximum time before cached content must be regenerated.
```

Comment:
Should we mention "from client or server"

Reply:
mmm I see, context matters here, who's regenerating this content, in reaction to what